core/src/main/java/com/novoda/noplayer/mediaplayer/AndroidMediaPlayerFacade.java

```diff
@@ -246,64 +249,18 @@ private boolean isInPlaybackState() {
                 && currentState != STATE_PREPARING;
     }
 
-    public void stop() {
+    void stop() {
         if (hasPlayer()) {
             mediaPlayer.stop();
         }
     }
 
     List<PlayerAudioTrack> getAudioTracks() {
-        if (mediaPlayer == null) {
-            throw new NullPointerException("You can only call getAudioTracks() when video is prepared.");
-        }
-
-        return getOnlyAudioTracks();
-    }
-
-    private ArrayList<PlayerAudioTrack> getOnlyAudioTracks() {
-        ArrayList<PlayerAudioTrack> audioTracks = new ArrayList<>();
-        for (MediaPlayer.TrackInfo trackInfo : mediaPlayer.getTrackInfo()) {
-            if (trackInfo.getTrackType() == MediaPlayer.TrackInfo.MEDIA_TRACK_TYPE_AUDIO) {
-                audioTracks.add(new PlayerAudioTrack(String.valueOf(trackInfo.hashCode()), trackInfo.getLanguage(), "", -1, -1));
-            }
-        }
-        return audioTracks;
-    }
-
-    void selectAudioTrack(int audioTrackIndex) {
-        if (mediaPlayer == null) {
-            throw new NullPointerException("You can only call selectAudioTrack() when video is prepared.");
-        }
-
-        int absoluteAudioTrackIndex = getAbsoluteAudioTrackIndex(audioTrackIndex);
-
-        if (absoluteAudioTrackIndex != INVALID_AUDIO_TRACK_INDEX) {
-            mediaPlayer.selectTrack(absoluteAudioTrackIndex);
-        }
+        return trackSelector.getAudioTracks(mediaPlayer);
     }
 
-    private int getAbsoluteAudioTrackIndex(int relativeAudioTrackIndex) {
-        int absoluteAudioTrackIndex = 0;
-
-        for (MediaPlayer.TrackInfo trackInfo : mediaPlayer.getTrackInfo()) {
-            if (trackInfo.getTrackType() == MediaPlayer.TrackInfo.MEDIA_TRACK_TYPE_AUDIO) {
-                absoluteAudioTrackIndex++;
-            }
-
-            if (absoluteAudioTrackIndex == relativeAudioTrackIndex) {
-                return absoluteAudioTrackIndex;
-            }
-        }
-
-        Log.e(String.format(
-                "Attempt to %s has been ignored because an invalid position was specified: %s, total: %s",
-                "getAbsoluteAudioTrackIndex()",
-                relativeAudioTrackIndex,
-                absoluteAudioTrackIndex
-              )
-        );
-
-        return INVALID_AUDIO_TRACK_INDEX;
+    void selectAudioTrack(PlayerAudioTrack playerAudioTrack) {
+        trackSelector.selectAudioTrack(mediaPlayer, playerAudioTrack);
```

Comment:
Pass the media player so it is in scope for the shortest amount of time otherwise we have a cyclic dependency where the facade has both the `trackSelector` and the `mediaPlayer` and where the `trackSelector` has the `mediaPlayer`.